ai/src/test/kotlin/ktx/ai/LeafTasksTest.kt

```diff
@@ -0,0 +1,144 @@
+package ktx.ai
+
+import com.badlogic.gdx.ai.btree.BehaviorTree
+import com.badlogic.gdx.ai.btree.Task
+import io.kotlintest.mock.mock
+import org.junit.Assert
+import org.junit.Test
+
+class LeafTasksTest {
+  @Test
+  fun `failureLeaf function should add Failure to BehaviorTree`() {
+    val tree = BehaviorTree<Cat>()
+    val initialChildCount = tree.childCount
+
+    tree.failureLeaf()
+
+    Assert.assertEquals(initialChildCount + 1, tree.childCount)
```

Comment:
Use asserts consistently (either static import of individual functions, or access with the `Assert` prefix - I lean towards the former).

Reply:
Good points. I'll clean these up and implement the rest of the functions.